problem_2_3.R

```diff
@@ -0,0 +1,59 @@
+# Calling various libraries
+library(tidyverse)
+
+###############
+## PROBLEM 2 ##
+###############
+
+# Importing the text data into a vector
+raw_file <- readLines(con = "http://www.sao.ru/lv/lvgdb/article/suites_dw_Table1.txt")
+
+# Getting the line number of the separator line
+sep_line = grep("^--", raw_file,TRUE)
+
+# Getting a header of the table by using sep_line as a reference for the header line
+header <- read.table(text = raw_file[sep_line-1], nrows = 1, header = FALSE, sep ='|',strip.white = TRUE)
+
+# Getting the data of the table and skipping the separator line 
+data <- read.table(text = raw_file,skip = sep_line, header = FALSE, sep = "|", strip.white = TRUE)
+
+# Adding the  header to the data
+colnames(data) <- header
+
+# Storing the descriptions into a separate text file
+write.table(raw_file[1:(sep_line-2)], # Using sep_line as a reference for the interval of lines
+            file = "suites_dw_descriptions.txt", sep = "\t",row.names = FALSE, quote=FALSE)
+
+###############
+## PROBLEM 3 ##
+###############
+
+# Plotting the distribution of the m_b variable which tells us the absolute magnitude of the galaxy
+# the lower the value the bigger is the magnitude of the respective galaxy
+
+data %>% 
+  ggplot(aes(x=m_b)) + 
+  geom_histogram(aes(y=..density..), # Histogram with density instead of count on y-axis
+                 binwidth=.5,
+                 colour="black", fill="white") +
+  geom_density(alpha=.2, fill="#FF6666") +
+  geom_vline(aes(xintercept=mean(m_b, na.rm=T)),# Adding mean and ignoring NA values
+             color="red", linetype="dashed", size=1) +
+  ggtitle("Distribution of the m_b variable") +
+  xlab("m_b (Absolute magnitude, lower value = bigger magnitude)")
+
+# It can be seen that that the observations of the variable are much less dense in the -5,0 interval
+# of observations which does point at the under-representation of the smaller magnitude galaxies,
+# furthermore the distribution of the variable seems to be skewered to the right and does not seem to
+# follow a normal distribution Which is confirmed when we run the Shapiro-Wilk test where we can 
+# reject h0 (w = 0.975, p < 0.01)
+shapiro.test(data$m_b)
+
+# If we look at the quantiles the median has a value of -13.73 and the 3rd quantile lies at the value of
+# -11,90,which coincides with the lack of smaller magnitude observations. 
+summary(data$m_b)
+
+# Since the magnitude of the galaxies is measured in the luminosity they emit the lack of lower
+# magnitude galaxies could be a result of being "hidden" by larger and brighter galaxies.
+ 
```

Comment:
You could have plotted a scatter plot with mass vs distance. The plot would reveal a tendency that small galaxies are observed more frequently close to us than far away from us. This may indicate that small galaxies are hard to notice when they are further away.